### DISCUSSION TITLE
Cannot query downsampled data in thanos

### DISCUSSION THREAD

**Discussion by rkdutta** [thanos-io/thanos#7102] [Questions & Answers]
I have the following Thanos compactor settings.

- "--retention.resolution-raw=2d"
- "--retention.resolution-5m=10d"
- "--retention.resolution-1h=20d"

I have also enabled --query.auto-downsampling flag in the Thanos query component. 
Still, I can't see data beyond 2 days. 

I have been investigating this issue for a long time, but I still have no luck.

**Comment by yeya24** [comment]
Hey, sorry for the late reply. I think the issue here is that in order to have your data to be downsampled to 5m, you need to have blocks for at least 40h length (max time - min time), which usually result in a block with 48h length.

This requires that you keep your raw resolution block for a longer time. Similar reason for 1h downsampled block, you need to have block with at least 10 day length which result in a block with 14 day length. So I recommend you keep your raw data for at least 15 days  to make sure they can be downsampled to 1h blocks
